src/amberscript/parser_run_test.cc

```diff
@@ -608,6 +608,46 @@ RUN my_pipeline DRAW_ARRAY AS TRIANGLE_LIST)";
   EXPECT_EQ(3U, cmd->GetVertexCount());
 }
 
+TEST_F(AmberScriptParserTest, RunDrawArraysIndexed) {
+  std::string in = R"(
+SHADER vertex my_shader PASSTHROUGH
+SHADER fragment my_fragment GLSL
+# GLSL Shader
+END
+BUFFER vtex_buf DATA_TYPE vec3<float> DATA
+1 2 3
+4 5 6
+7 8 9
+END
+
+PIPELINE graphics my_pipeline
+  ATTACH my_shader
+  ATTACH my_fragment
+  VERTEX_DATA vtex_buf LOCATION 0
+END
+
+RUN my_pipeline DRAW_ARRAY AS TRIANGLE_LIST INDEXED)";
```

Comment:
The pipeline doesn't actually have an INDEX_DATA.  It seems we could validate that and return an error if the pipeline doesn't actually have an index buffer attached to it.

Reply:
Good idea. Done.